app/src/components/NavBar.vue

```diff
@@ -79,17 +104,67 @@
   </div>
   <!-- </header> -->
 </template>
+
 <script setup lang="ts">
 import { logout } from '@/utils/navBarUtil'
 import IconHamburgerMenu from '@/components/icons/IconHamburgerMenu.vue'
 import IconBell from '@/components/icons/IconBell.vue'
 import { NETWORK } from '@/constant/index'
+import { ref, type Ref, watch } from 'vue'
+import { type NotificationResponse } from '@/types'
+import { useCustomFetch } from '@/composables/useCustomFetch'
 const emits = defineEmits(['toggleSideButton', 'toggleEditUserModal', 'withdraw'])
+
 defineProps<{
   withdrawLoading: boolean
   balanceLoading: boolean
   balance: string
 }>()
+
+const isUnread = ref(false)
+const id: Ref<number | string | null> = ref(null)
+
+const {
+  //isFetching: isNotificationsFetching,
+  //error: notificationError,
+  data: notifications,
+  execute: executeFetchNotifications
+} = useCustomFetch<NotificationResponse>('notification').json()
+
+const {
+  //isFetching: isUpdateNotificationsFetching,
+  //error: isUpdateNotificationError,
+  execute: executeUpdateNotifications
+  //data: _notifications
+} = useCustomFetch<NotificationResponse>('notification', {
+  immediate: false,
+  beforeFetch: async ({ options, url, cancel }) => {
+    //options.body = JSON.stringify({ id: id.value })
+    url += `?id=${id.value}`
+    return { options, url, cancel }
+  }
+})
```

Comment:
You can create a reactive variable 
const updateEndPoint=ref("")

```suggestion
} = useCustomFetch<NotificationResponse>(updateEndPoint, {immediate: false})
```

And when you update the value of update endpoint and call the executeUpdateNotification. 
The event will trigger.